### PR TITLE
VACMS-21879: operating status situation updates

### PR DESCRIPTION
**Pull request by brianseek**
# Description

Adds the situation updates from banner alerts to operation status pages

This pull request adds support for formatting and displaying "situation updates" for VAMC Operating Status and Alerts. It introduces new types, updates mock/test data, and implements logic to extract, sort, and render situation updates from banner alerts, including handling edge cases such as unpublished alerts and missing updates. The UI now displays these updates in a dedicated section.

**Situation Updates Feature**

* Added the `SituationUpdates` type and integrated it into the `VamcOperatingStatusAndAlerts` formatted type, enabling structured handling of updates and info fields.
* Implemented the `buildSituationUpdates` function in `query.ts` to extract, sort (by most recent date), and format situation updates from published banner alerts, including handling info-only cases.
* Updated the component (`template.tsx`) to render a new "Situation updates and information" section, displaying each update and associated info with correct date formatting and HTML content. [[1]](diffhunk://#diff-48846ebb7c8d75ed04f93d14176c4f790bb5625a507da7057794bab00ada1ce3R1-R16) [[2]](diffhunk://#diff-48846ebb7c8d75ed04f93d14176c4f790bb5625a507da7057794bab00ada1ce3L44-R95)

**Testing and Mock Data**

* Expanded unit tests to cover situation updates logic, including sorting, filtering unpublished alerts, and info-only scenarios.
* Enhanced mock data to include realistic situation updates and related fields for more robust testing and demonstration.

**Type and Integration Improvements**

* Updated types for banner alerts and related entities to support new fields and optionality, ensuring type safety and flexibility. [[1]](diffhunk://#diff-a5035ec5110b73d89f63d6fbe12f3bd0af37cf28d17289acfd0bdb8c0932ccbbR50) [[2]](diffhunk://#diff-a5035ec5110b73d89f63d6fbe12f3bd0af37cf28d17289acfd0bdb8c0932ccbbL120-R128)

These changes collectively provide a comprehensive solution for displaying timely situation updates on VAMC facility pages, improving both data handling and user experience.

## Ticket

Closes [21879](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/21879)

## Developer Task

- [ ] PR submitted against the `main` branch of `next-build`.
- [ ] Link to the issue that this PR addresses (if applicable).
- [ ] Define all changes in your PR and note any changes that could potentially be breaking changes.
- [ ] PR includes steps to test your changes and links to these changes in the Tugboat preview (if applicable).
- [ ] Provided before and after screenshots of your changes (if applicable).
- [ ] Alerted the #next-build Slack channel to request a PR review.
- [ ] You understand that once approved, you are responsible for merging your changes into `main`. (Note that changes to `main` will move automatically into production.)

## Testing Steps

This can't be tested easily on the tugboat for this PR.

- [ ] using a CMS tugboat, create a banner with situation updates for a facility
- [ ] point the next deployment to this branch and re-deploy
- [ ] visit page to verify

## Screenshots
localhost_3999_oklahoma-city-health-care_operating-status_ (3)

### The rare (should never happen) instance where there are two active banners with situation updates

localhost_3999_oklahoma-city-health-care_operating-status_ (2)

---

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository.

## Standard Checks

- [ ] Code Quality: Readabilty, Naming Conventions, Consistency, Reusability
- [ ] Test Coverage: 80% coverage
- [ ] Functionality: Change functions as expected with no additional bugs
- [ ] Performance: Code does not introduce performance issues
- [ ] Documentation: Changes are documented in their respective README.md files
- [ ] Security: Packages have been approved in the TRM

## Merging a Layout

When merging a layout, you must ensure that the content type has been turned on for `next-build` in the [.tugboat.env](../envs/.tugboat.env). This method mocks the CMS flag that must be turned on for a layout to be included in the build.

The layout component and matching resource type should be included in the [slug.tsx](../src/pages/[[...slug]].tsx), so that it can reviewed. Including a component in the slug.tsx does not mean a page will be viewable in production only on the tugboat for the branch.

When a layout is merged to main and approved for deployment, the prod CMS will turn the toggle on for the resource type. 

The status of layouts should be kept up to date inside [templates.md](../READMEs/templates.md). This includes QA progress, development progress, etc. A link should be provided for where testing can occur.

### CODE REVIEW

File: src/types/drupal/node.ts [hunk 1]
```diff
@@ -117,13 +118,14 @@ export interface NodeFullWidthBannerAlert extends DrupalNode {
   field_body: FieldFormattedText
   field_alert_dismissable: boolean
   field_alert_type: string
-  field_banner_alert_situationinfo: FieldFormattedText
+  field_banner_alert_situationinfo: string
```
Comment (pwolfert) [inline src/types/drupal/node.ts:121]: 👍
